packages/graphiql-2-rfc-context/src/api/providers/GraphiQLBrowserProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { parseQueryStringURL, updateQueryStringURL } from '@graphiql/toolkit';
+import { BrowserState } from '../types';
+
+import {
+  BrowserAction,
+  BrowserActionTypes,
+  queryStringParamsChangedAction,
+} from '../actions/browserActions';
+
+export type BrowserReducer = React.Reducer<BrowserState, BrowserAction>;
+export interface BrowserHandlers {
+  changeQueryStringParams: (parameter: string, value: string) => void;
+  dispatch: React.Dispatch<BrowserAction>;
+}
+
+export const initialBrowserState: BrowserState = {
+  queryStringParams: {
+    operation:
+      parseQueryStringURL(window.location.search).operation ||
+      parseQueryStringURL(window.location.search).query,
+    variables: parseQueryStringURL(window.location.search).variables,
+    operationName: parseQueryStringURL(window.location.search).operationName,
+  },
+  queryStringParamsURL: '',
```

Comment:
I'm not sure I understand why we need to have both `queryStringParams` and `queryStringParamsURL`. If I understand it correctly, they both contain the same data, just formatted differently.